### ISSUE TITLE
Remove Index.__inv__

### ISSUE THREAD

**Issue by toobaz** [pandas-dev/pandas#22335]
#### Code Sample, a copy-pastable example if possible

```python
In [1]: import pandas as pd

In [2]: idx = pd.Index([2, 4, 6])

In [3]: idx.__inv__()
Out[3]: Int64Index([-2, -4, -6], dtype='int64')
```
#### Problem description

``idx.__inv__`` does what ``idx.__neg__`` does, and not do what its name suggests (that is, ``operator.inv``). By the way, ``~idx`` raises ``TypeError`` anyway, and ``idx.__invert__`` does not exist, differently from what [the Python docs suggest](https://docs.python.org/2/library/operator.html).

Notice that instead
```
In [4]: s = pd.Series([2, 4, 6])

In [5]: ~s
Out[5]: 
0   -3
1   -5
2   -7
dtype: int64

In [6]: s.__invert__()
Out[6]: 
0   -3
1   -5
2   -7
dtype: int64
```
which is correct (and ``Series.__inv__`` does not exist).

#### Expected Output

```
Out[3]: Int64Index([-3, -5, -7], dtype='int64')
```

Given how wrong, undocumented and not exposed this is, I suggest to immediately remove ``Index.__inv__``. Then we can later decide to introduce ``Index.__invert__`` which does the right thing, and also enable the ``~`` operator.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.0-6-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: it_IT.UTF-8
LOCALE: it_IT.UTF-8

pandas: 0.24.0.dev0+460.geb0ac5437
pytest: 3.5.0
pip: 9.0.1
setuptools: 39.2.0
Cython: 0.28.4
numpy: 1.14.3
scipy: 0.19.0
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.5.6
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.4
blosc: None
bottleneck: 1.2.0dev
tables: 3.3.0
numexpr: 2.6.1
feather: 0.3.1
matplotlib: 2.2.2.post1634.dev0+ge8120cf6d
openpyxl: 2.3.0
xlrd: 1.0.0
xlwt: 1.3.0
xlsxwriter: 0.9.6
lxml: 4.1.1
bs4: 4.5.3
html5lib: 0.999999999
sqlalchemy: 1.0.15
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: 0.2.1
gcsfs: None

</details>

**Comment by jorisvandenbossche** [comment]
> By the way, ~idx raises TypeError anyway

Do you know why that is? Because in principle, if `__inv__()` works, so should `~` (that might indicate another bug in our code)

The TypeError is raised by the `Index.__inv__` implementation, but `Int64Index` overrides that with a working implementation.

**Comment by toobaz** [comment]
> Do you know why that is? Because in principle, if __inv__() works, so should ~ (that might indicate another bug in our code)

To the best of my understanding, and at least from Python 2.7 onwards, ``~`` is overridden just by using ``__invert__``, not ``__inv__`` (although the latter is exposed by the ``operators`` module).

**Comment by jbrockmendel** [comment]
We define `__invert__` in 5 places, `__inv__` in one.  Is one of these wrong?

**Comment by toobaz** [comment]
Given that ``__inv__`` has the wrong name (for what it does), and that the name is of an obscure method (not supported by any other pandas object, not sure it even exists in numpy, and at the same time trivial to compute as ``1/idx``), I think this bug should be fixed by:
- making a copy of the method at https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexes/base.py#L5429 with the name changed to ``__invert__``, so that at least the right method works
- insert a deprecation warning in the original ``__inv__`` telling users to switch to ``__invert__`` (or ``~``), and later delete it

**Comment by toobaz** [comment]
By the way, I'm not even sure a method which only exists as ``__invert__`` requires a deprecation cycle...

**Comment by jbrockmendel** [comment]
closed by #45006
